source/common/stats/stats_impl.h

```diff
@@ -443,8 +495,16 @@ template <class Base> class IsolatedStatsCache {
  */
 class IsolatedStoreImpl : public Store {
 public:
-  IsolatedStoreImpl()
-      : counters_([this](const std::string& name) -> CounterSharedPtr {
+  // TODO(jmarantz): It appears that if the configuartion overrides the max
+  // option length, this isolated store will not see the adjustment. I am
+  // not sure if that's a problem. Ideally I'd like to take the stat_options
+  // here we can creating the StatAllocator, but it's not immediately obvious
+  // where to pull that from. I think this issue existed starting with
+  // https://github.com/envoyproxy/envoy/pull/3629, which removed the shadowing
+  // of option size in a static.
+
+  /* explicit */ IsolatedStoreImpl(/*Stats::StatsOptions& stats_options*/)
```

Comment:
Note behavior regression here:

- prior to #3629 : truncated the same as other stats due to shared static max length
- after #3629 : not truncated at all (AFAICT)
- with this PR : truncated to default length (127 chars).

If someone (@ambuc @mattklein123 @mrice32) can share the desired behavior (not truncated at all vs truncated with the other stats) I will fix.

Reply:
I think this comes down to the question of whether or not we expect an IsolatedStore to have stats which are interoperable with other stats. My instinct is no, due to its isolated nature -- in which case this behavior is not something anyone relies upon. Thoughts?

Reply:
Per discussion, I force the truncation at 1M bytes, so effectively we are not truncating in this PR.